### ISSUE TITLE
relay/generated-typescript-types false positive when the type is defined in a separate file

### ISSUE THREAD

**Issue by Hypnosphi** [relayjs/eslint-plugin-relay#165]
Reproduction: https://github.com/Hypnosphi/eslint-relay-issue, run `npm install && npm run lint`

```
/src/App.tsx
  13:16  error  The prop passed to useFragment() should be typed with the type 'AppFragment$key' imported from 'AppFragment.graphql', e.g.:

  import type {AppFragment$key} from 'AppFragment.graphql';  relay/generated-typescript-types

✖ 1 problem (1 error, 0 warnings)

```

It works fine if I move the `Props` definition from `App.types.ts` to `App.tsx`

**Comment by captbaritone** [comment]
I think this is working as expected. Defining your props locally within the component that uses those props is aligned with the goal of colocation, which this lint rule is intended to encourage. In other words, your module should explicitly define its dependencies locally within the same module. That includes defining the props type locally.
